includes/admin/upgrades/v3/class-data-migrator.php

```diff
@@ -657,12 +657,56 @@ public static function orders( $data = null ) {
 
 		$set_tax_rate_meta = false;
 
-		if ( ! empty( $tax_rate ) && ! empty( $order_address_data['country'] ) ) {
```

Comment:
I think I would revert all of the changes made here. With the global added correctly to `edd_get_tax_rate_by_location`, if a migrated order has the global tax rate applied, it's assigned automatically. If the tax rate doesn't match the retrieved tax rate, we have just added it to the order as metadata, and that happens automatically too.

Creating an inactive global rate to assign to an order would just be confusing, I think, and be inconsistent with how we are handling old orders with a country or regional tax rate which differs from the current country/regional rate. In those cases, we are adding the tax rate as order meta; we should do the same for the global rate.

Since the `$order_address_data['region']` is definitely set here, although it may be an empty string, instead of completely reverting, we could update the code to be less redundant:

```php
			$tax_rate_object = edd_get_tax_rate_by_location(
				array(
					'country' => $order_address_data['country'],
					'region'  => $order_address_data['region'],
				)
			);
```

Reply:
I can understand the logic of if we find an order that doesn't match the currently set regional tax rate (or global now) that we just add meta, and not an inactive rate. However I do think that we should, if we find an order that is not a regional order, but matches the fallback tax rate, assigns to the newly created default tax rate ID.

Reply:
It would do that--basically whatever tax rate that's found that matches will get assigned to the order, as long as the tax rate amount matches. The behavior in 3.0 currently is to match orders with existing (imported) tax rates--with the fix in `edd_get_tax_rate_by_location` then the global could end up matching--and assign the ID if there is a match; otherwise the rate is added as metadata. Reverting to this makes that possible with the global rate (at least that's what it's doing for me locally; the code changes were never assigning the default rate.

The current iteration is working for orders, but for an order which doesn't match an existing tax rate but does have tax collected, no inactive tax rate is being added to the database, and I would expect for it to be.

I think part of the issue is that when `$tax_rate_object` is defined in line 663 and 699, both of those are using a function which returns an array of tax rate objects, not a single tax rate object, so the metadata is assigned anyway and an inactive tax rate is not created. Since I don't think one should be created, I'm fine with that, but just noting that the code is not working the way I think it's intended as written.

Reply:
@robincornett We'll have to update `edd_get_tax_rate_by_location()` if we take this approach, as that function returns early if the `country` is empty:

```
function edd_get_tax_rate_by_location( $args ) {

	$rate = false;
	if ( empty( $args['country'] ) ) {
		return $rate;
	}
```

Reply:
Possibly, but I think we force any store collecting taxes to collect address data, and the country/region are required. This `$args['country']` is coming from the order address, so any order with tax should have a country. If it doesn't, that feels like a different issue?

Reply:
I've removed it in 8b3ca3a95aa233d9e39b65d7f0613220847173a4 and it seems that everything seems to work as I expect, will you give that a try @robincornett. I just moved the empty check to around the first search and that allows it to go down to search further.